### PR TITLE
Wagtail 4

### PR DESCRIPTION
**Pull request by katdom13**
Release notes: https://docs.wagtail.org/en/stable/releases/4.0.html

- Updated docs, test to pass for Wagtail 4

Tests are OK

```
Found 17 test(s).
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
.................
----------------------------------------------------------------------
Ran 17 tests in 0.481s

OK
```

### CODE REVIEW

**Review by nickmoreton — COMMENTED** [review]
Just one suggestion here.

File: tests/settings.py [hunk 1]
```diff
@@ -96,3 +96,8 @@
 }
 
 DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
+
+
+# This should be the base URL used to access the Wagtail admin site.
+# Without this, URLs in notification emails will not display correctly.
+WAGTAILADMIN_BASE_URL = "http://example.com"
```
[the remark this answers is not in the document]
Comment (gasman) [reply tests/settings.py]: It's only _required_ for Wagtail >=3, but it won't do any harm on older versions - I don't think we'll gain anything from putting it in a conditional.

[the remark this answers is not in the document]
Comment (katdom13) [reply tests/settings.py]: Thank you for this. I have placed this inside a conditional:
https://github.com/wagtail/wagtail-generic-chooser/pull/58/commits/878ab08a89c2b18044805b71568a39484fa2fc00

**Comment by katdom13** [comment]
Hi @gasman ,
I would like to ask for further review on this. If there are none, can this be merged?
Thank you!

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ It is intended that `wagtail-generic-chooser` will be expanded to cover all the
 
 ## Requirements
 
-Wagtail 2.4 or higher
+Wagtail 2.15 or higher
```
Comment (gasman) [inline README.md:11]: I don't think we need to change this, as we're not breaking anything against older versions (even if they're out of support).

File: tests/settings.py [hunk 3]
```diff
@@ -96,3 +96,9 @@
 }
 
 DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
+
+# Suppresses warning for versions of Wagtail above 3.0
+if WAGTAIL_VERSION >= (3, 0):
```
Comment (gasman) [inline tests/settings.py:101]: As mentioned earlier I don't think this conditional adds any value - `WAGTAILADMIN_BASE_URL` is a valid setting on any version, even if it only raises a warning on >=3.0. No worries - I'll roll this back when I merge.

**Review by gasman — APPROVED** [review]
Thanks @katdom13! Have now had a chance to test the updated package against Wagtail 4, and while the styling seems to have degraded slightly due to the CSS changes in 4.x, it's still fully functional, so I'll get this merged and put a new release out on Monday.

**Comment by gasman** [comment]
Merged in a8440cb687cab8441540235bcee06b86c6551b71.

**Pull request closed without merging**